File: 1288. Remove Covered Intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var removeCoveredIntervals = function(intervals) {

    intervals.sort((first, second) => {

        if (first[0] === second[0]) {
            return second[1] - first[1];
        }

        return first[0] - second[0];
    });

    let maxEnd = -1;

    let remaining = 0;

    for (const [start, end] of intervals) {

        if (end > maxEnd) {

            remaining++;

            maxEnd = end;
        }
    }

    return remaining;
};

File: 1288. Remove Covered Intervals.java
class Solution {

    public int removeCoveredIntervals(
            int[][] intervals
    ) {

        Arrays.sort(
                intervals,
                (first, second) -> {

                    if (first[0] == second[0]) {
                        return second[1] - first[1];
                    }

                    return first[0] - second[0];
                }
        );

        int maxEnd = -1;

        int remaining = 0;

        for (int[] interval : intervals) {

            if (interval[1] > maxEnd) {

                remaining++;

                maxEnd = interval[1];
            }
        }

        return remaining;
    }
}
